src/Domain/CoordinateMaps/TimeDependent/ShapeMapTransitionFunctions/Wedge.hpp

```diff
@@ -0,0 +1,252 @@
+ * gradient of $f$ becomes
+ *
+ * \begin{equation}
+ * \frac{\partial f}{\partial x_i} = \frac{\frac{\partial
+ * D_{\text{out}}}{\partial x_i} - \frac{x_i}{r}}{D_{\text{out}} -
+ * D_{\text{in}}} - \frac{\left(D_{\text{out}} - r\right) - \left(\frac{\partial
+ * D_{\text{out}}}{\partial x_i} \right)}{\left(D_{\text{out}} -
+ * D_{\text{in}}\right)^2}.
+ * \end{equation}
+ *
+ * If the outer boundary is spherical (but not the inner boundary), then the
+ * gradient of $f$ becomes
+ *
+ * \begin{equation}
+ * \frac{\partial f}{\partial x_i} = \frac{-\frac{x_i}{r}}{D_{\text{out}} -
+ * D_{\text{in}}} - \frac{\left(D_{\text{out}} - r\right) + \left(\frac{\partial
+ * D_{\text{in}}}{\partial x_i} \right)}{\left(D_{\text{out}} -
+ * D_{\text{in}}\right)^2}
+ * \end{equation}
+ *
+ * If the both boundaries are spherical, then the gradient of $f$ simplifies
+ * greatly to
+ *
+ * \begin{equation}
+ * \frac{\partial f}{\partial x_i} = \frac{-\frac{x_i}{r}}{D_{\text{out}} -
+ * D_{\text{in}}}
+ * \end{equation}
+ */
+class Wedge final : public ShapeMapTransitionFunction {
+  struct Surface {
+    double radius{};
+    double sphericity{};
+
+    // This is the distance from the center (assumed to be 0,0,0) to this
+    // surface in the same direction as coords.
+    template <typename T>
+    T distance(const std::array<T, 3>& coords) const;
+
+    void pup(PUP::er& p);
+
+    bool operator==(const Surface& other) const;
+    bool operator!=(const Surface& other) const;
+  };
+
+ public:
+  explicit Wedge() = default;
+
+  /*!
+   * \brief Construct a Wedge transition for a wedge block in a given direction.
+   *
+   * \details Many concentric wedges can be part of the same falloff from 1 at
+   * the inner boundary of the innermost wedge, to 0 at the outer boundary of
+   * the outermost wedge.
+   *
+   * \param overall_inner_radius Inner radius of innermost wedge
+   * \param overall_outer_radius Outermost radius of outermost wedge
+   * \param overall_inner_sphericity Sphericity of innermost surface of
+   * innermost wedge
+   * \param overall_outer_sphericity Sphericity of outermost surface of
+   * outermost wedge
+   * \param this_wedge_inner_radius This particular wedge's inner radius
+   * \param this_wedge_outer_radius This particular wedge's outer radius
+   * \param this_wedge_inner_sphericity Sphericity of this particular wedge's
+   * inner surface
+   * \param this_wedge_outer_sphericity Sphericity of this particular wedge's
+   * outer surface
+   * \param orientation_map The same orientation map that was passed into the
+   * corresponding `domain::CoordinateMaps::Wedge` block.
+   */
+  Wedge(double overall_inner_radius, double overall_outer_radius,
+        double overall_inner_sphericity, double overall_outer_sphericity,
+        double this_wedge_inner_radius, double this_wedge_outer_radius,
+        double this_wedge_inner_sphericity, double this_wedge_outer_sphericity,
```

Comment:
Question: what do you need the "this wedge" quantities for? I thought the transition only depends on the inner and outer radius/sphericity.

Reply:
For the inverse function, we have to return `nullopt` if we are outside of the given block so we need to know this block's bounds

Reply:
That's not a requirement we currently impose on maps. Many maps return logical coordinates outside of [-1, 1] from the inverse function and only return `nullopt` if they failed to do the inverse.

Reply:
Question that's tangentially related, do we actually use the inverse mapping at all in the evolution equations? Or is is just for horizon finds/control system things/observing in different frames?

Reply:
The inverse is only used for things like interpolations I think.

Reply:
Hmm ok I'll change it to just check if it's inside the range of the transition function not the block itself

Reply:
@nilsvu ok pushed a fixup